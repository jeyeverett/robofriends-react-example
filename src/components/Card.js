import React from 'react'; //Because we are writing JSX, we always need to import react in these files

//Below we use the functional way of building a component instead of class-based
//Note that we can only return ONE element, i.e. everything needs to be inside of one parent element - this used to mean we put everything in a div, but now we can use <React.Fragment></React.Fragment> or just <></>
//Note that we can destructure props inside the function parameter field i.e. {name, email, id} in the parameter field is equivalent to const {name, email, id} = props;
const Card = ({name, email, id}) => { 
    return (
        <div className='bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5 tc'>
            <img src={`https://robohash.org/${id}?200x200`} alt='' />
            <div>
                <h2>{name}</h2>
                <p>{email}</p>
            </div>
        </div>
    )
}

export default Card;