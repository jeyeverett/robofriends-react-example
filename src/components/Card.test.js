import { create } from "react-test-renderer";
import React from 'react';
import Card from './Card';

const card = create(<Card />);

it('expect to render Card component', () => {
    expect(card.toJSON()).toMatchSnapshot()
});