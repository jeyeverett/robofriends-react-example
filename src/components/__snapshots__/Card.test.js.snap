// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<div
  className="bg-green dib br3 pa4 ma2 grow bw2 shadow-5 tc"
>
  <img
    alt="RoBro"
    height="200"
    src="https://robohash.org/undefined?size=200x200&set=set3"
    width="200"
  />
  <div>
    <h2
      className="ttu white f4"
    />
    <p
      className="ttu white i"
    />
  </div>
</div>
`;
